release/tests/test_github_api.py

```diff
@@ -0,0 +1,156 @@
+"""
+Tests for testeng-ci/release/github_api
+"""
+import ddt
+import json
+import re
+import responses
+from responses import GET, POST  # pylint: disable=no-name-in-module
+from unittest import TestCase
+
+from release.tests.aborted import Aborted
+from release.github_api import RequestFailed, GithubApi
+
+
+class EndpointInfo(object):
+    # pylint: disable=too-few-public-methods
+    """ Encapsulates the info we need to test an end point """
+
+    def __init__(
+            self,
+            expected_url,
+            request_maker,
+            success=200,
+            method=GET,
+            request_body=None
+    ):  # pylint: disable=too-many-arguments
+        self.expected_url = expected_url
+        self.request_maker = request_maker
+        self.success = success
+        self.method = method
+        self.request_body = request_body
+
+ENDPOINTS = [
+    EndpointInfo(
+        "https://api.github.com/repos/test-org/test-repo/commits",
+        GithubApi.commits
```

Comment:
GithubApi.commits is just a one argument function that takes an API, so it should be equivalent to ``lambda api: api.commits()`` and it's a little shorter this way and IMO clearer since it's obvious where the commits method lives.

That said, it is a little less polymorphic here, since it means we have to refer directly to the API class. Happy to change it if you think it's better or more consistent that way.